### PR TITLE
Transport process

### PR DESCRIPTION
**Pull request by rmsrosa**
This introduces a `NoiseTransport` type to create noise random processes of the form `W_t = f(t, RV)`, where `RV` is a random variable or random vector.

### CODE REVIEW

**Comment by rmsrosa** [comment]
Ok, this is it. Some examples of the API can be seen in the docstring for the `NoiseTransport` type.

Some comments.

1. Currently, solving a RODE or an SDE with a `NoiseTransport` noise does not give a new realization of the random variable defining the noise. It is necessary to make a small change in `solve.jl` over at `StochasticDiffEq.jl` to check for a `NoiseTransport` noise and reseed it (currently only a `NoiseProcess` is reseeded). But this change needs `NoiseTransport` to exist, so I can make a PR there once/if this is approved.

2. A `NoiseFunction` can be disguised as a `NoiseTransport` by making `f(u, t, p, Y)` not depend on `Y` and by doing `RV(rng) = nothing` (or whatever). It is tempting to simply add this functionality to `NoiseFunction`, but, as it is, we would have an ambiguity when providing `Z`. We could change the order of the arguments for `NoiseTransport` to make things smooth and non-breaking, but it would be a bit cumbersome to add `nothing` for `Z` explicitly when one really wants a `NoiseTransport`. And it would be unnecessary stuff if one just wants a `NoiseFunction`. It might be best to just keep them separately.

3. Concerning that thing with `W = prob.noise` when solving an `AbstractRODEProblem`, this is used to `reinit!` a noise, which is a good thing, despite the side effects. For a `NoiseProblem`, however, we currently have `W = deepcopy(prob.noise)`, so the solver never knows whether that is the first solve of a problem or not, so `reinit!` doesn't work the same. To compensate for that, I force a `reinit!` regardless, so we cannot define the first realization of the random variable, unfortunately.  I am not sure that is the best thing to do.

File: test/noise_transport.jl [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        out[1] = sin(v[1] * t)
+        out[2] = sin(t + v[2])
+        out[3] = cos(t) * v[1] + sin(t) * v[2]
+        nothing
+    end
+
+    RV!(rng, v) = (v[1] = randn(rng); v[2] = rand(rng))
+    rv = zeros(2)
+
+    @test_nowarn NoiseTransport(t0, f!, RV!, rv, noise_prototype = zeros(3))
```
Comment (frankschae) [inline test/noise_transport.jl:53]: Should we maybe have a test with respect to `NoiseFunction` to ensure that if both can be used they provide the same result.

**Review by frankschae — APPROVED** [review]
This looks good to me. I agree that having `NoiseTransport` and `NoiseFunction` separately is cleaner/easier to handle. 

I also don't have a better solution for the `reinit!` issue. I think triggering `reinit!` every time is at least a safe option.

File: test/noise_transport.jl (hunk 1 above)
Comment (rmsrosa) [reply test/noise_transport.jl:53]: > Should we maybe have a test with respect to NoiseFunction to ensure that if both can be used they provide the same result.

Yeah, good idea. I will do that.

Comment (rmsrosa) [reply test/noise_transport.jl:69]: Done!🙂

**Pull request merged**